types/jest/index.d.ts

```diff
@@ -315,17 +328,79 @@ declare namespace jest {
 
     interface MockInstance<T> {
         mock: MockContext<T>;
+        /** 
+         * Clears all information stored in the mockFn.mock.calls and mockFn.mock.instances arrays.
+         * 
+         * Often this is useful when you want to clean up a mock's usage data between two assertions.
+         */
         mockClear(): void;
+        /** 
+         * Resets all information stored in the mock, including any inital implementation given. 
+         * 
+         * This is useful when you want to completely restore a mock back to its initial state.
+         */
         mockReset(): void;
+        /**
+         * Removes the mock and restores the initial implementation.
+         * 
+         * This is useful when you want to mock functions in certain test cases and restore the original implementation in others.
+         */
+        mockRestore(): void;
+        /**
+         * Accepts a function that should be used as the implementation of the mock. 
+         * 
+         * The mock itself will still record all calls that go into and instances that come from itself – the only difference is that the implementation will also be executed when the mock is called.
+         * 
+         * Note: jest.fn(implementation) is a shorthand for mockImplementation.
+         * 
+         * @param {Function} fn The initola implmentation
+         * @returns {Mock<T>} A Jest mock
+         */
         mockImplementation(fn: Function): Mock<T>;
+        /**
+         * Accepts a function that will be used as an implementation of the mock for one call to the mocked function. Can be chained so that multiple function calls produce different results.
+         * 
+         * @example
+         * var myMockFn = jest.fn(() => 'default')
+         *   .mockImplementationOnce(() => 'first call')
+         *   .mockImplementationOnce(() => 'second call');
+
+         * console.log(myMockFn(), myMockFn(), myMockFn(), myMockFn());
+         * > 'first call', 'second call', 'default', 'default'
+         * 
+         * @param {Function} fn A factory function to run the next time a mock is called
+         * @returns {Mock<T>} The same Jest mock
+         */
         mockImplementationOnce(fn: Function): Mock<T>;
+        /**
+         * Just a simple sugar function for: `jest.fn( () => this)`.
+         * 
+         * @returns {Mock<T>} The same Jest mock
+         */
         mockReturnThis(): Mock<T>;
+        /**
+         * @deprecated Use ` jest.fn(() => value` instead.
```

Comment:
I was able to get some clarification (via https://github.com/facebook/jest/pull/3993 being merged) that this is not actually deprecated. The documentation for `mockReturnValueOnce` is updated as well. https://github.com/facebook/jest/blob/master/docs/en/MockFunctionAPI.md#mockfnmockreturnvaluevalue